### PR TITLE
[ENH] Removed `NB_IS_FEDERATION_API` environment variable

### LINKED ISSUE #239: Remove `NB_IS_FEDERATION_API` environment variable
We've decided to no longer make the API type that the query tool points to configurable, so this variable is no longer needed.

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #239 

Changes proposed in this pull request:

- Removed code related to `NB_IS_FEDERATION_API`
- Updated workflow config files
- Updated README.md

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,16 +77,15 @@ To set environment variables, create a `.env` file in the root directory and add
 
 ```bash
 NB_API_QUERY_URL=http://localhost:8000/
-NB_IS_FEDERATION_API=false # For node API
 ```
 
-if you're using the remote (in this example federation) api, your `.env` file would look something like this:
```
Comment (surchs) [inline README.md:88]: good find

File: README.md [hunk 2]
```diff
@@ -82,16 +77,15 @@ To set environment variables, create a `.env` file in the root directory and add
...
+if you're using the remote api, your `.env` file would look something like this:
 
 ```bash
 NB_API_QUERY_URL=https://federate.neurobagel.org/
 ```
 
-if you're using a federation api with authentication, your `.env` file would look something like this:
+if you're using a remote api with authentication, your `.env` file would look something like this:
```
Comment (surchs) [inline README.md:88]: in this case, why not keep `federation`?  we now know it has to always be a f-API.

**Review by surchs — APPROVED** [review]
Thanks for the PR @rmanaem, really nice to see how much simpler everything is getting with this change!

Very :cherries: but I find `[ENH]` a bit odd for this. Technically we're taking a feature away. How about `[MNT]` instead? Or maybe we need a new prefix for feature deprecation / removal?

I left a comment, and then :cook:

File: README.md (hunk 2 above)
Comment (rmanaem) [reply README.md:88]: Since we only support `federation`, the condition here (if) no longer makes sense.

**Pull request merged**